src/Search/searchTools.js

```diff
@@ -0,0 +1,118 @@
+// @flow
+import uniq from "lodash/uniq";
+import flatten from "lodash/flatten";
+import type { QueryType } from "./Types/QueryType";
+
+// function which normalize input values
+export const transformString = (str: string): string => {
+  return str
+    .toLowerCase()
+    .replace(/ /g, "")
+    .normalize("NFD")
+    .replace(/[\u0300-\u036f]/g, "");
+};
+
+// function which compares data table value with searched value in terms of type and operator
+type FilterByColNameAndOp = {
+  comparedValue: string | number,
+  operator: string,
+  searchedValue: string
+};
+
+export const filterByColNameAndOp = ({
+  comparedValue,
+  operator,
+  searchedValue
+}: FilterByColNameAndOp): boolean => {
+  let res: boolean = false;
+  let formattedOp: string = operator;
+  if (formattedOp === "=") {
+    formattedOp = "==";
+  }
+  switch (typeof comparedValue) {
+    case "boolean":
+      if (formattedOp === "==" || formattedOp === "!=") {
+        return eval(`${comparedValue.toString()} ${formattedOp} ${searchedValue}`);
+      }
+      break;
+    case "number":
+      return eval(
+        `${comparedValue} ${formattedOp} ${parseInt(searchedValue.replace(/ /g, ""), 10)}`
+      );
+    case "string":
+      if (formattedOp === "==" || formattedOp === "!=") {
+        // const res = transformString(comparedValue) === transformString(searchedValue);
+        res = transformString(comparedValue).includes(transformString(searchedValue));
+      }
+      if (formattedOp === "!=") {
+        return !res;
+      }
+      return res;
+    default:
+      return res;
+  }
+  return false;
+};
+
+// Research function when you're not in query mode
+type SimpleSearch = {
+  columns: Object[],
+  value: string,
+  rows: Object[]
+};
+
+export const simpleSearch = ({ columns, value, rows }: SimpleSearch): Object[] => {
+  const resRows: Object[] = [];
+  rows.forEach((row: Object) => {
+    columns.some(
+      ({ id }: { id: string }) =>
+        filterByColNameAndOp({
```

Comment:
Not necessary function called, simply use `transformString(comparedValue).includes(transformString(searchedValue));`